### DISCUSSION TITLE
Can OAuth Apps making LLM requests with the user's credentials?

### DISCUSSION THREAD

**Discussion by CNSeniorious000** [community/community#135292] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

Can a GitHub OAuth App request GitHub Models as the user itself (if the user permits this)?

**Comment by srijon57** [comment, marked as answer]
Yes, a GitHub OAuth App can make requests on behalf of the user if the user authorizes it to do so. When a user authorizes an OAuth App, the app receives an access token that represents the user's permissions. This token allows the app to make API requests as if the user were making them directly.In the context of GitHub Models or any other GitHub resources, if the OAuth App has been granted the appropriate scopes during the authorization process, it can request those resources on behalf of the user. For example, if the app needs to access repositories or interact with issues, it would need the relevant scopes like repo or read:user.

**Reply by CNSeniorious000** [reply]
Thanks!
